backend/tests/integration/records/api/test_api_literature.py

```diff
@@ -1190,13 +1190,17 @@ def test_adding_record_with_document_without_filename(inspire_app, s3):
             "key": expected_document_key,
             "url": current_s3_instance.get_public_url(expected_document_key),
             "original_url": "http://original-url.com/2",
-            "filename": "key",
+            "filename": expected_document_key,
```

Comment:
I think this is wrong, you're completing discarding the `key` that is provided on L1180 here. So the test doesn't have a problem, but your code probably does.